File: src/Java/myservicebus/src/main/java/com/myservicebus/serialization/RawJsonSerializerFactory.java
package com.myservicebus.serialization;

import com.fasterxml.jackson.databind.ObjectMapper;

public final class RawJsonSerializerFactory implements SerializerFactory {
    private final MessageHeaderConvention headerConvention;
    private final ObjectMapper mapper;

    public RawJsonSerializerFactory() {
        this(JsonSerializationDefaults.createObjectMapper(), MassTransitHeaderConvention.INSTANCE);
    }

    public RawJsonSerializerFactory(MessageHeaderConvention headerConvention) {
        this(JsonSerializationDefaults.createObjectMapper(), headerConvention);
    }

    public RawJsonSerializerFactory(ObjectMapper mapper) {
        this(mapper, MassTransitHeaderConvention.INSTANCE);
    }

    public RawJsonSerializerFactory(ObjectMapper mapper, MessageHeaderConvention headerConvention) {
        if (mapper == null) {
            throw new IllegalArgumentException("mapper must not be null");
        }
        this.mapper = mapper;
        this.headerConvention = headerConvention;
    }

    @Override
    public String getContentType() {
        return DefaultInboundMessageResolver.RAW_JSON_CONTENT_TYPE;
    }

    @Override
    public MessageSerializer createSerializer() {
        return new RawJsonMessageSerializer(mapper, headerConvention);
    }

    @Override
    public MessageDeserializer createDeserializer() {
        return new RawJsonMessageDeserializer(mapper, headerConvention);
    }
}

File: src/MyServiceBus/Serialization/RawJsonSerializerFactory.cs
using System.Text.Json;

namespace MyServiceBus.Serialization;

public sealed class RawJsonSerializerFactory : ISerializerFactory
{
    private readonly IMessageHeaderConvention _headerConvention;
    private readonly JsonSerializerOptions _jsonSerializerOptions;

    public RawJsonSerializerFactory()
        : this(JsonSerializationDefaults.CreateOptions(), MassTransitHeaderConvention.Instance)
    {
    }

    public RawJsonSerializerFactory(IMessageHeaderConvention headerConvention)
        : this(JsonSerializationDefaults.CreateOptions(), headerConvention)
    {
    }

    public RawJsonSerializerFactory(
        JsonSerializerOptions jsonSerializerOptions,
        IMessageHeaderConvention? headerConvention = null)
    {
        ArgumentNullException.ThrowIfNull(jsonSerializerOptions);
        _jsonSerializerOptions = jsonSerializerOptions;
        _headerConvention = headerConvention ?? MassTransitHeaderConvention.Instance;
    }

    public string ContentType => InboundMessageResolver.RawJsonContentType;

    public IMessageSerializer CreateSerializer() => new RawJsonMessageSerializer(_jsonSerializerOptions, _headerConvention);

    public IMessageDeserializer CreateDeserializer() => new RawJsonMessageDeserializer(_jsonSerializerOptions, _headerConvention);
}
